### PR TITLE
Automatically collapse navbar menu when clicking nav item

### LINKED ISSUE #317: RFE: Auto collapse Navbar during navigation
Recollapsing the nav-bar when a user clicks on a link would be a great feature enhancement.  Hooking into the ember router `didTransition` event should be possible to accomplish this.

### PR DESCRIPTION
**Pull request by simonihmig**
Closes #317

### CODE REVIEW

**Comment by simonihmig** [comment]
The Navbar's behavior is changed here, as clicking a nav item will now automatically collapse the navbar's content (on mobile). I don't know if that should be regarded a breaking change, or rather a bug fix, as this is how the original Bootstrap Navbar works when used in a non-SPA?

File: addon/components/base/bs-navbar.js [hunk 1]
```diff
@@ -42,6 +42,20 @@ const { observer } = Ember;
  * `collapse`: triggers the `onCollapse` action and collapses the navbar (mobile)
  * `expand`: triggers the `onExpand` action and expands the navbar (mobile)
 
+ ### Responsive Design
+
+ For the mobile breakpoint the Bootstrap styles will hide the navbar content (`{{navbar.content}}`). Clicking on the
+ navbar toggle button (`{{navbar.toggle}}`) will expand the menu. By default all NavItems (`{{nav.item}}`) are already
+ wired up to call the navbar's `collapse` action, so clicking any of them will collapse the navbar. To selectively
+ prevent that, you can override the NavItem's `onClick` action, with a custom action if needed, or just be setting it
```
Comment (srvance) [inline addon/components/base/bs-navbar.js]: Edit: "or just set it"

**Comment by danidr** [comment]
I would like to point out two things:

1. Nav item could be used for other elements than links; therefore, there are many realistic cases where the onClick should be set to false manually. This would also break backwards compatibility.
2. Nav Item in bootstrap has a padding to avoid having link text overlapping the border, for example in case of dropdown menus. This would bring a case where clicking on an item would collapse the menu (since the item registers the onClick action) but the actual link wouldn't trigger because the user has clicked just a wee bit outside of it, therefore confusing the user as to what has happened.

**Comment by simonihmig** [comment]
@danidr I took your feedback into account, and changed the API accordingly, see https://github.com/kaliber5/ember-bootstrap/pull/365/files#diff-a0ee6ab773729a34ca7901616a846cbe. 

@danidr @srvance Mind having another look?

**Comment by danidr** [comment]
@simonihmig If nobody has any comment about this implementation, I would say it looks good to me!

**Pull request merged**